### PR TITLE
[HYDRATOR-595]: fix Action API for PipelineConfigurer

### PR DESCRIPTION
**Pull request by thegreenfrog**
Issue: https://issues.cask.co/browse/HYDRATOR-595

### CODE REVIEW

File: cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/co/cask/cdap/etl/api/action/Action.java [hunk 1]
```diff
@@ -30,4 +31,14 @@
    * @throws Exception when there is failure in method execution
    */
   public abstract void run(ActionContext context) throws Exception;
+
+  /**
+   * Implement this method to validate configurations
```
Comment (anwar6953) [inline cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/co/cask/cdap/etl/api/action/Action.java]: nit - have a space before the `@param` line

Comment (anwar6953) [inline cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/co/cask/cdap/etl/api/action/Action.java]: Do we need a javadoc for this? The method defined in the interface already has a javadoc, right

Comment (albertshau) [reply cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/co/cask/cdap/etl/api/action/Action.java]: yeah its not needed

Comment (thegreenfrog) [reply cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/co/cask/cdap/etl/api/action/Action.java]: oh yeah good point.

**Comment by knazir** [comment]
Does this also require a change in `PipelineAction` to call `action.configurePipeline(...)`? If I recall correctly, this method was not implemented for actions because `configurePipeline` is not called (no schemas for actions?). Could you point me to an example of an action where a user has to override it @anwar6953 ?

**Comment by thegreenfrog** [comment]
but don't we use `configurePipeline` as the place to validate configs before the action runs though @knazir ?

**Comment by knazir** [comment]
Yeah that's true @thegreenfrog, my bad kind of misunderstood the change.

**Comment by anwar6953** [comment]
lgtm

**Comment by thegreenfrog** [comment]
guild is breen: http://builds.cask.co/browse/CDAP-DUT4725-2

**Pull request merged**